torch/distributed/_tensor/ops/_matrix_ops.py

```diff
@@ -360,6 +362,20 @@ def scaled_dot_product_efficient_attention_strategy(
     ]
     if has_attn_bias:
         all_replicate.append(Replicate())  # attn bias
+
+    # Context Parallelism: shards on the sequence dim
+    single_mesh_dim_strategies.append(
```

Comment:
I thought about this more, I think we should not include the context parallel strategy to the sharding strategy list here by default. The reason is that: context parallel strategy (shard on seq dim) is not a mathmatically correct strategy without ring attention being called locally. But if we add the strategy here, then the inputs with the strategy listed here would go through without ring attention being called, which could result in the wrong numerical results.

Reply:
good point. I agree we should not have the strategy by default. But I think we still need the propagation rule so that once we integrate with TP, the output sharding can be correctly inferred. I will have a separate PR to fix this.